python/lsst/ts/standardscripts/system_wide_shutdown.py

```diff
@@ -0,0 +1,266 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["SystemWideShutdown"]
+
+import asyncio
+import re
+import types
+import typing
+
+import yaml
+from lsst.ts.idl import get_idl_dir
+
+from lsst.ts import salobj
+
+
+class SystemWideShutdown(salobj.BaseScript):
+    """Discover all running CSCs and send them all to OFFLINE state.
+
+    Notes
+    -----
+
+    This SAL Script works by getting a list of components from the
+    """
+
+    def __init__(self, index: int) -> None:
+        super().__init__(index=index, descr="Send all CSCs to OFFLINE.")
+        self.idl_file_pattern_match = re.compile(
+            r"(.*)sal_revCoded_(?P<component>.*).idl"
+        )
+        self.failed: dict[str, str] = dict()
+        self.components_to_ignore = ["Script", "ScriptQueue"]
+        self.components_to_end_with = ["ScriptQueue"]
+
+    @classmethod
+    def get_schema(cls) -> None | dict[str, typing.Any]:
+        schema_yaml = """
+$schema: http://json-schema.org/draft-07/schema#
+$id: https://github.com/lsst-ts/ts_standardscripts/python/lsst/ts/standardscripts/system_wide_shutdown.py
+title: SystemWideShutdown v1
+description: Configuration for SystemWideShutdown script.
+type: object
+properties:
+    user:
+        description: Name of the user that is executing the script.
+        type: string
+    reason:
+        description: Reason for running the system wide shutdown.
+        type: string
+    ignore:
+        description: CSCs to ignore.
+        type: array
+        items:
+            type: string
+        default: []
+    start_with:
+        description: CSCs to start with.
+        type: array
+        items:
+            type: string
+        default: []
+    end_with:
+        description: CSCs to end with.
+        type: array
+        items:
+            type: string
+        default: []
+required: [user, reason]
+additionalProperties: false
+        """
+        return yaml.safe_load(schema_yaml)
+
+    async def configure(self, config: types.SimpleNamespace) -> None:
+        self.config = config
+
+    def set_metadata(self, metadata: salobj.type_hints.BaseMsgType) -> None:
+        metadata.duration = 60.0
+
+    async def run(self) -> None:
+
+        components_running = await self.discover_components()
+
+        self.log.info(
+            f"Shutting down {len(components_running)} as requested by {self.config.user}. "
```

Comment:
Maybe change to f"Shutting down {len(components_running)} `components` as requested by {self.config.user}. "?